spec.html

```diff
@@ -34622,38 +34637,144 @@ <h1>%TypedArray%.prototype.filter ( _callbackfn_ [ , _thisArg_ ] )</h1>
+          1. If _n_ is +&infin;, return *-1*<sub>𝔽</sub>.
+          1. Else if _n_ is -&infin;, set _n_ to 0.
+          1. If _n_ &ge; 0, then
+            1. Let _k_ be _n_.
+          1. Else,
+            1. Let _k_ be _len_ + _n_.
+            1. If _k_ &lt; 0, set _k_ to 0.
+          1. Repeat, while _k_ &lt; _len_,
+            1. Let _kPresent_ be ! HasProperty(_O_, ! ToString(𝔽(_k_))).
+            1. If _kPresent_ is *true*, then
+              1. Let _elementK_ be ! Get(_O_, ! ToString(𝔽(_k_))).
+              1. Let _same_ be the result of performing Strict Equality Comparison _searchElement_ === _elementK_.
+              1. If _same_ is *true*, return 𝔽(_k_).
+            1. Set _k_ to _k_ + 1.
+          1. Return *-1*<sub>𝔽</sub>.
+        </emu-alg>
+        <p>This function is not generic. The *this* value must be an object with a [[TypedArrayName]] internal slot.</p>
       </emu-clause>
 
       <emu-clause id="sec-%typedarray%.prototype.join">
         <h1>%TypedArray%.prototype.join ( _separator_ )</h1>
-        <p>%TypedArray%`.prototype.join` is a distinct function that implements the same algorithm as `Array.prototype.join` as defined in <emu-xref href="#sec-array.prototype.join"></emu-xref> except that the *this* value's [[ArrayLength]] internal slot is accessed in place of performing a [[Get]] of *"length"*. The implementation of the algorithm may be optimized with the knowledge that the *this* value is an object that has a fixed length and whose <emu-xref href="#integer-index">integer-indexed</emu-xref> properties are not sparse. However, such optimization must not introduce any observable changes in the specified behaviour of the algorithm.</p>
-        <p>This function is not generic. ValidateTypedArray is applied to the *this* value prior to evaluating the algorithm. If its result is an abrupt completion that exception is thrown instead of evaluating the algorithm.</p>
+        <p>The interpretation and use of the arguments of %TypedArray%`.prototype.join` are the same as for `Array.prototype.join` as defined in <emu-xref href="#sec-array.prototype.join"></emu-xref>.</p>
+        <p>When the `join` method is called with one argument _separator_, the following steps are taken:</p>
+        <emu-alg>
+          1. Let _O_ be the *this* value.
+          1. Perform ? ValidateTypedArray(_O_).
+          1. Let _len_ be _O_.[[ArrayLength]].
+          1. If _separator_ is *undefined*, let _sep_ be the single-element String *","*.
+          1. Else, let _sep_ be ? ToString(_separator_).
+          1. Let _R_ be the empty String.
+          1. Let _k_ be 0.
+          1. Repeat, while _k_ &lt; _len_,
+            1. If _k_ &gt; 0, set _R_ to the string-concatenation of _R_ and _sep_.
+            1. Let _element_ be ! Get(_O_, ! ToString(𝔽(_k_))).
+            1. If _element_ is *undefined* or *null*, let _next_ be the empty String; otherwise, let _next_ be ! ToString(_element_).
```

Comment:
Is it actually be possible for `_element_` to be `*null*`? It looks to me like `_O_` is always an integer-indexed exotic object and `ToString(𝔽(_k_))` is always a [CanonicalNumericIndexString](https://tc39.es/ecma262/#sec-canonicalnumericindexstring), which means the above `Get` is [going to](https://tc39.es/ecma262/#sec-integer-indexed-exotic-objects-get-p-receiver) go through [IntegerIndexedElementGet](https://tc39.es/ecma262/#sec-integerindexedelementget), which can only return numerics or `*undefined*`.

If that's correct, the `or *null*` here should be removed, since it's misleading.